evalml/pipelines/components/utils.py

```diff
@@ -380,3 +381,27 @@ def make_balancing_dictionary(y, sampling_ratio):
             # this class is already larger than the ratio, don't change
             class_dic[index] = value_counts[index]
     return class_dic
+
+
+def downcast_int_nullable_to_double(X):
+    """Downcasts IntegerNullable types to Double in order to support certain estimators like ARIMA, CatBoost, and LightGBM.
+
+    Args:
+        X (pd.DataFrame): Feature data.
+
+    Returns:
+        X: DataFrame initialized with logical type information where IntegerNullables are cast as Double.
+
+    """
+    if not isinstance(X, pd.DataFrame):
+        return X
+    if X.ww.schema is None:
+        X.ww.init()
```

Comment:
Probably better to use `infer_feature_types` here instead, for consistency's sake

Reply:
For some reason, I can't reply to your comment on `test_automl.py`, but to address that here - the `check_dtype=False` fixes the time series regression paramaterization of the test.  The problem here, I believe, is that Woodwork inference is reinfering the results of the predictions, which happen to come out to be "0.0" as "0", due to the issue that Parthiv filed.  With the latest woodwork, the output of our predictions are integers, which I think is really weird for time series regression.